src/tools/rbd/action/Trash.cc

```diff
@@ -273,6 +280,29 @@ int do_list(librbd::RBD &rbd, librados::IoCtx& io_ctx, bool long_flag,
         break;
     }
 
+    cls::rbd::TrashImageSpec trash_spec;
+    r = librbd::cls_client::trash_get(&io_ctx, entry.id, &trash_spec);
```

Comment:
I'm not sure I like the idea of breaking the librbd API barrier here by directly interacting w/ the OSDs via a cls method. It might not even be worth the effort to expose the state outside of librbd since this is really just an odd transient edge condition.